aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -301,23 +302,28 @@ void ImagingResampleVertical(
 // weights, but when aa=False they could be optimized further.
 template <typename scale_type, class F>
 void upsample_avx_bilinear_uint8(
-    const at::Tensor& input,
+    const at::Tensor& input_,
     const at::Tensor& output,
     bool align_corners,
     const scale_type& scales,
     bool antialias) {
-  auto batch_size = input.size(0);
-  auto num_channels = input.size(1);
-  auto xin = input.size(3);
-  auto yin = input.size(2);
+  auto batch_size = input_.size(0);
+  auto num_channels = input_.size(1);
+  auto xin = input_.size(3);
+  auto yin = input_.size(2);
   auto xout = output.size(3);
   auto yout = output.size(2);
 
   if (xin == xout && yin == yout) {
-    output.copy_(input);
+    output.copy_(input_);
     return;
   }
 
+  at::Tensor input = input_;
+  if (!(input.is_contiguous() || input.is_contiguous(at::MemoryFormat::ChannelsLast))) {
+    input = input.contiguous(at::MemoryFormat::ChannelsLast);
```

Comment:
are we sure this will always create a copy? IIRC from offline discussions, the python equivalent needed `copy=True` to make sure the result was properly `channels_last`. Maybe that's not a problem with the C++ version?

Reply:
The other discussion was about `tensor.to(memory_format=..., copy=True)` method and `tensor.contiguous()` does not have `copy` parameter

Reply:
`tensor.contiguous(mf)` calls `tensor.clone(mf)` if tensor is not already contiguous with mf and clone does:
- `empty_like`
- `copy_`

https://github.com/pytorch/pytorch/blob/b3b333205fb0362d1f303a4c1054592c5fa4aea3/aten/src/ATen/native/TensorProperties.cpp#L93-L102

https://github.com/pytorch/pytorch/blob/b3b333205fb0362d1f303a4c1054592c5fa4aea3/aten/src/ATen/native/TensorFactories.cpp#L1593-L1614